### ISSUE TITLE
UX Improvement - change king order display logic

### ISSUE THREAD

**Issue by f3mshep** [somethiiing/appvalon#38]
It has been suggested that displaying the king order using a queue is not the best UX. By changing the king order by "physically" moving the players around (popping the previous king off the queue, shifting her to end of the queue) instead of displaying order under the player names breaks continuity. 

This could involve changing just the frontend, but the data structure returned from the backend could be changed as well (playerObj could have a kingOrder property, for example)

**Comment by bbailey27** [comment]
@f3mshep I'm unclear on how a frontend-only solution would work. I feel like the server state would at least need to store a startingKingOrder or something similar. What was your plan for this?

**Comment by f3mshep** [comment]
@bbailey27 

TL;DR - 
1. make backend never change the kingOrder, and have frontend rely only on the existing isKing property in each playerObj to display who is king
2. Clean up the data structure a little bit. Delete the KingOrder array property, give each playerObj an immutable tableOrder id. Use isKing property in playerObj to display who is king.

------

1. We would use some sort of cursor in the UI. Cursor placement would be calculated based on who is first in the King queue.  Backend would still update the queue as normal. The problem with that approach is it requires the frontend persisting something, and does not rely on the backend to handle store logic. If we wanted to do something quick, we could just have the backend never change the kingOrder, and have the frontend rely on the isKing property to manage the king cursor. You are correct that this would still entail a tiny bit of backend work. 

My thinking is a preferred solution would include the following:

2. PlayerObj gets an immutable table order property. Frontend displays players based on that property. We already have an isKing property - this would handle the cursor. Backend would get rid of the kingOrder array, since it would be logic duplication. This would require a frontend change in addition to backend, but the data structure would be a little cleaner IMO.

**Comment by bbailey27** [comment]
I don't think having the frontend persist the original king order will work out without how we have things structured to rely on the backend. I think having the backend never change player order would be ideal. No need to duplicate different player lists for display, just one static list and we can use the isKing property. We already use the isKing property to control most frontend king-specific logic.

So backend task:

- Don't rotate kingOrder - keep a static player order list and update the isKing player properties when the crown advances (I feel like this list should belong to the room not the PlayerObj but I may be unclear on naming)

And frontend tasks:

- Remove all references to kingOrder
- Convert anywhere relying on kingOrder[0] to use .isKing instead
- Replace uses of kingOrder for display with the new static list

**Comment by f3mshep** [comment]
I am working on the backend portion of this. If no one else wants the frontend bit, I can take that too.

**Comment by f3mshep** [comment]
I just realized, we will need to update hammer logic as well.
